Lib/test/test_code.py

```diff
@@ -428,6 +436,21 @@ def func():
             self.assertIsNone(line)
             self.assertEqual(end_line, new_code.co_firstlineno + 1)
 
+    def test_large_lnotab(self):
+        d = {}
+        lines = (
+            ["def f():"] +
+            ['    """'] +
+            ['    .'] * (1 << 17) +
+            ['    """'] +
+            ["    x = 1"] * (1 << 17)
```

Comment:
I don't think you need to use a docstring for this. Blank lines should be fine:

```suggestion
            [""] * (1 << 17) +
            ["    pass"] * (1 << 17)
```

This has the added benefit of not failing when run under `-OO`!